### PR TITLE
Update conflict constraint for "nelmio/api-doc-bundle" in order to allow version 3

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject
Update conflict constraint for "nelmio/api-doc-bundle" in order to allow version 3.

I am targeting this branch, because these changes almost respect BC (with one, very little BC-break potential).

## Changelog

```markdown
### Added
- support for `nelmio/api-doc-bundle` >= 3.6
```

### CODE REVIEW

File: src/Controller/Api/Legacy/MessageController.php
Comment (phansys) [inline src/Controller/Api/Legacy/MessageController.php:80]:
```suggestion
        $criteria = array_filter(static function($value): bool {
            return null !== $value;
        }, $criteria);
```

**Review by phansys — CHANGES_REQUESTED** [review]

File: src/Controller/Api/Legacy/MessageController.php
Comment (wbloszyk) [reply src/Controller/Api/Legacy/MessageController.php:80]: input must be first
```suggestion
$criteria = array_filter($criteria, static function ($value): bool {
            return null !== $value;
        });
```

**Comment by wbloszyk** [comment]
@phansys 
Do you think we can merge this PRs? or I should wait for other reviews?

File: UPGRADE-3.x.md
Comment (phansys) [inline UPGRADE-3.x.md:7]:
```suggestion
```

**Comment by phansys** [comment]
> Do you think we can merge this PRs? or I should wait for other reviews?

I think this is RTM, but I generally wait for at least 2 approvals, if there are not controversial changes with active rejections.

**Review by phansys — APPROVED** [review]

File: UPGRADE-3.x.md
Comment (phansys) [inline UPGRADE-3.x.md:13]:
```suggestion
Since version 3.x, support for nelmio/api-doc-bundle > 3.6 is added. Controllers for NelmioApiDocBundle v2 were moved under `Sonata\NotificationBundle\Controller\Api\Legacy\` namespace and controllers for NelmioApiDocBundle v3 were added as replacement. If you extend them, you must ensure they are using the corresponding inheritance.
```

**Review by phansys — CHANGES_REQUESTED** [review]

**Review by phansys — APPROVED** [review]

**Review by phansys — COMMENTED** [review]
Tests are [failing](https://github.com/sonata-project/SonataNotificationBundle/pull/478/checks?check_run_id=1151378379#step:11:89).

**Comment by wbloszyk** [comment]
@phansys I know. I need UserBundle release to drop support for datagrid <3 first. then I can remove support for datagrid < 3.0 here.

**Comment by wbloszyk** [comment]
@sonata-project/contributors It is RTM. Please review it.

File: UPGRADE-3.x.md
Comment (phansys) [inline UPGRADE-3.x.md:13]:
```suggestion
Controllers for NelmioApiDocBundle v2 were moved under `Sonata\NotificationBundle\Controller\Api\Legacy\` namespace and controllers for NelmioApiDocBundle v3 were added as replacement. If you extend them, you MUST ensure they are using the corresponding inheritance.
```

**Review by phansys — APPROVED** [review]

File: UPGRADE-3.x.md
Comment (wbloszyk) [reply UPGRADE-3.x.md]: Done.

**Review by phansys — APPROVED** [review]

**Review by VincentLanglet — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Thanks

**Pull request merged**
